src/descriptor/key.rs

```diff
@@ -560,10 +576,10 @@ impl<'secp, C: secp256k1::Verification> Copy for DescriptorPublicKeyCtx<'secp, C
 
 impl<'secp, C: secp256k1::Verification> DescriptorPublicKeyCtx<'secp, C> {
     /// Create a new context
-    pub fn new(secp_ctx: &'secp secp256k1::Secp256k1<C>, child_number: bip32::ChildNumber) -> Self {
+    pub fn new(secp_ctx: &'secp secp256k1::Secp256k1<C>, index: u32) -> Self {
         Self {
             secp_ctx: secp_ctx,
-            child_number: child_number,
+            index: index,
```

Comment:
Should we panic here or return a result in order to avoid the panic of index >= 2^31 when calling derive.

Reply:
I think we should panic. It's very easy for the caller to sanity-check that their numbers are in range, and very annoying to have to unwrap in the 99% of cases when you know you're in range.